src/main/java/ch/njol/skript/util/EnumUtils.java

```diff
@@ -28,90 +28,93 @@
 import java.util.HashMap;
 import java.util.Locale;
 
+/**
+ * A language utility class to be used for easily handling language values representing an Enum.
+ * @param <E> Generic representing the Enum.
+ * @see ch.njol.skript.classes.EnumClassInfo
+ */
 public final class EnumUtils<E extends Enum<E>> {
 	
-	private final Class<E> c;
+	private final Class<E> enumClass;
 	private final String languageNode;
 
+	@SuppressWarnings("NotNullFieldNotInitialized") // initialized in constructor's refresh() call
 	private String[] names;
 	private final HashMap<String, E> parseMap = new HashMap<>();
 	
-	public EnumUtils(Class<E> c, String languageNode) {
-		assert c != null && c.isEnum() : c;
-		assert languageNode != null && !languageNode.isEmpty() && !languageNode.endsWith(".") : languageNode;
+	public EnumUtils(Class<E> enumClass, String languageNode) {
+		assert enumClass.isEnum() : enumClass;
+		assert !languageNode.isEmpty() && !languageNode.endsWith(".") : languageNode;
 		
-		this.c = c;
+		this.enumClass = enumClass;
 		this.languageNode = languageNode;
 
-		names = new String[c.getEnumConstants().length];
+		refresh();
 		
-		Language.addListener(() -> validate(true));
+		Language.addListener(this::refresh);
 	}
 	
 	/**
-	 * Updates the names if the language has changed or the enum was modified (using reflection).
+	 * Refreshes the representation of this Enum based on the currently stored language entries.
 	 */
-	void validate(boolean force) {
-		boolean update = force;
-
-		E[] constants = c.getEnumConstants();
+	void refresh() {
+		E[] constants = enumClass.getEnumConstants();
+		names = new String[enumClass.getEnumConstants().length];
+		parseMap.clear();
```

Comment:
this should work too right?
```suggestion
		E[] constants = enumClass.getEnumConstants();
		names = new String[constants.length];
		parseMap.clear();
```